File: datastruct-algoritem-java/src/main/java/datastruct/queue/DequeueTest.java
package datastruct.queue;

/**
 * @author czx
 * @date 2019/12/24
 */
public class DequeueTest {
    public static void main(String[] args) {
        Dequeue<Integer> dq = new Dequeue<>();
        for (int i = 0; i < 10; i++) {
            dq.push(i);
        }
        dq.forEach(e -> System.out.print(e + " "));
        System.out.println("dq.size() = " + dq.size());
        System.out.println();
        dq.pushFront(100);
        dq.pushFront(200);
        dq.forEach(e -> System.out.print(e + " "));
        System.out.println("dq.size() = " + dq.size());
        dq.pop();
        dq.popFront();
        dq.forEach(e -> System.out.print(e + " "));
        System.out.println("dq.size() = " + dq.size());
    }
}

File: datastruxt-algoritem-go/datastruct/queue/dequeue_test.go
package queue

import (
	"fmt"
	"testing"
)

// Create by czx on 2019/12/25

func TestQueue(t *testing.T) {
	dq := NewDequeue()
	for i := 0; i < 10; i++ {
		dq.Push(i)
	}
	dq.Foreach(func(v interface{}) {
		fmt.Printf("%v ", v)
	})
	fmt.Println()
	fmt.Println("size: ", dq.Size())
	peek := dq.Peek()
	fmt.Println("peek value: ", peek)
	if ok, i := dq.Pop(); ok {
		fmt.Println("pop success: ", i)
	}
	if ok, i := dq.Pop(); ok {
		fmt.Println("pop success: ", i)
	}
	dq.Clear()
	fmt.Println("queue is empty: ", dq.Empty())

	for i := 0; i < 10; i++ {
		dq.PushFront(i)
	}
	dq.Foreach(func(v interface{}) {
		fmt.Printf("%v ", v)
	})
	fmt.Println("size: ", dq.Size())
	if ok, i := dq.PopFront(); ok {
		fmt.Println("pop front: ", i)
	}
	if ok, i := dq.Pop(); ok {
		fmt.Println("pop back: ", i)
	}
}
